langserver/references.go

```diff
@@ -212,6 +212,32 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 		t := time.NewTicker(time.Second)
 		defer t.Stop()
 		streamTick = t.C
+		streamPos := 0
+		streamUpdate = func() {
+			mu.Lock()
+			partial := refs
+			mu.Unlock()
+			if len(partial) == streamPos {
```

Comment:
What happens if `partial` (which is really `refs` cloned) changes behind the scenes? Is returning here important?